app/src/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -199,6 +200,74 @@ public static String getMd5Hash(File file) {
 
     }
 
+    public static String getExtension(String filePath) {
+        if (filePath.contains(".")) {
+            return last(filePath.split("\\."));
+        }
+        return "";
+    }
+
+    /**
+     * Get the last element of a String array.
+     */
+    private static String last(String[] strings) {
+        return strings[strings.length - 1];
+    }
+
+    /**
+     * Attempts to scale down an image file based on the max dimension given. If at least one of
+     * the dimensions of the original image exceeds that maximum, then make the larger side's
+     * dimension equal to the max dimension, and scale down the smaller side such that the
+     * original aspect ratio is maintained
+     */
+    public static boolean scaleImage(File originalImage, String finalFilePath, int maxDimen) {
+        boolean scaledImage = false;
+        String extension = getExtension(originalImage.getAbsolutePath());
+        ImageWidget.ImageType type = ImageWidget.ImageType.fromExtension(extension);
+        if (type == null) {
+            // The selected image is not of a type that can be decoded to or from a bitmap
+            Log.i(t, "Could not scale image " + originalImage.getAbsolutePath() + " due to incompatible extension");
```

Comment:
Yep we use the original, because that's what happens when scaleImage returns false.